app/scripts/lib/environment.js

```diff
@@ -64,9 +64,16 @@
 
     isFramed: function () {
       var win = this.window;
-      // about:accounts places FxA into an iframe, but is not considered
-      // to use the iframe flow. about:accounts sets window.name = 'remote'
-      return !! (win.top && win.top !== win && win.name !== 'remote');
+      return !! (
+                 win.top && win.top !== win &&
+                 // about:accounts places FxA into an iframe, but is not
+                 // considered to use the iframe flow. about:accounts
+                 // sets window.name = 'remote'
+                 win.name !== 'remote' &&
+                 // Mozilla Payments uses a trusted iframe
+                 // with window.name = 'payflow'. See #2607
+                 win.name !== 'payflow'
```

Comment:
I wonder if we should `console.log` or something when one of these hacky checks triggers, to assist debugging for far-future users who accidentally collide with the window-naming scheme here.